h/search/query.py

```diff
@@ -271,7 +277,7 @@ def nipsa_filter(group_service, user=None):
     # If any one of these "should" clauses is true then the annotation will
     # get through the filter.
     should_clauses = [
-        {"not": {"term": {"nipsa": True}}},
+        {"bool": {"must_not": {"term": {"nipsa": True}}}},
```

Comment:
Old way docs: https://www.elastic.co/guide/en/elasticsearch/reference/1.5/query-dsl-not-filter.html
New way docs: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-bool-query.html

Reply:
Concern: if the query is cached and then the user is shadowban it might not actually shadowban them. It says in the docs that it will rebuild the query results on changes to es indexing so as soon as the nipsa change get's indexed it should rebuild the query results and shadowban them.

Reply:
> so as soon as the nipsa change get's indexed it should rebuild the query results and shadowban them.

Yes, that's my understanding. The cached filter should be automatically invalidated and discarded when necessary without any user intervention.